content/en/docs/tasks/traffic-management/egress/egress-gateway-tls-origination/index.md

```diff
@@ -631,38 +571,64 @@ to hold the configuration of the NGINX server:
     $ kubectl create -n istio-system secret generic nginx-ca-certs --from-file=nginx.example.com/2_intermediate/certs/ca-chain.cert.pem
     {{< /text >}}
 
-1.  Generate the `istio-egressgateway` deployment with a volume to be mounted from the new secrets. Use the same options
-    you used for generating your `istio.yaml`:
+1.  To include a volume mounted from the new created secret, update the `istio-egressgateway` deployment.
+    To patch the `istio-egressgateway` deployment, create the following `gateway-patch.json` file:
 
     {{< text bash >}}
-    $ istioctl manifest generate --set values.gateways.istio-ingressgateway.enabled=false \
-    --set values.gateways.istio-egressgateway.enabled=true \
-    --set 'values.gateways.istio-egressgateway.secretVolumes[0].name'=egressgateway-certs \
-    --set 'values.gateways.istio-egressgateway.secretVolumes[0].secretName'=istio-egressgateway-certs \
-    --set 'values.gateways.istio-egressgateway.secretVolumes[0].mountPath'=/etc/istio/egressgateway-certs \
-    --set 'values.gateways.istio-egressgateway.secretVolumes[1].name'=egressgateway-ca-certs \
-    --set 'values.gateways.istio-egressgateway.secretVolumes[1].secretName'=istio-egressgateway-ca-certs \
-    --set 'values.gateways.istio-egressgateway.secretVolumes[1].mountPath'=/etc/istio/egressgateway-ca-certs \
-    --set 'values.gateways.istio-egressgateway.secretVolumes[2].name'=nginx-client-certs \
-    --set 'values.gateways.istio-egressgateway.secretVolumes[2].secretName'=nginx-client-certs \
-    --set 'values.gateways.istio-egressgateway.secretVolumes[2].mountPath'=/etc/nginx-client-certs \
-    --set 'values.gateways.istio-egressgateway.secretVolumes[3].name'=nginx-ca-certs \
-    --set 'values.gateways.istio-egressgateway.secretVolumes[3].secretName'=nginx-ca-certs \
-    --set 'values.gateways.istio-egressgateway.secretVolumes[3].mountPath'=/etc/nginx-ca-certs > \
-    ./istio-egressgateway.yaml
+    $ cat > gateway-patch.json <<EOF
```

Comment:
Why change from using `istioctl manifest generate` to patch? Does it not work? Seems like we should make sure we are in agreement on what is the recommended best practice for adding certs to a gateway.

Reply:
@frankbu this is for the file mounted approach only - and `istioctl manifest` does not work for the array objects.  I followed what was done for ingress file mount approach: https://istio.io/docs/tasks/traffic-management/ingress/secure-ingress-mount/#redeploy-istio-ingressgateway-with-the-new-certificate
Ingress PR/discussion: https://github.com/istio/istio.io/pull/4222